plugins/inventory/vmware_vm_inventory.py

```diff
@@ -624,6 +645,19 @@ def _get_hostname(self, properties, hostnames):
             if hostname:
                 return to_text(hostname)
 
+    def _can_add_host(self, host_filters, host_properties, host, strict=False):
+        for host_filter in host_filters:
+            try:
+                can_add_host = self._compose(host_filter, host_properties)
+            except Exception as e:  # pylint: disable=broad-except
+                if strict:
+                    raise AnsibleError("Could not evaluate %s as host_filters %s" % (host_filter, to_native(e)))
+                display.warning(to_native(e))
+
+            if not can_add_host:
```

Comment:
If you give an invalid template, it fails everything with this:

```
[WARNING]:  * Failed to parse /tmp/awx_172_nd86rma3/vmware_vm_inventory.yml
with auto plugin: local variable 'can_add_host' referenced before assignment
```

Reply:
Done. Thanks.